### PR TITLE
Report fulfillment errors in new trait solver

### PR DESCRIPTION
**Pull request by compiler-errors**
Causes fewer ICEs when testing the new solver :smile:

### CODE REVIEW

File: compiler/rustc_trait_selection/src/solve/fulfill.rs [hunk 1]
```diff
@@ -48,29 +51,46 @@ impl<'tcx> TraitEngine<'tcx> for FulfillmentCtxt<'tcx> {
...
-            unimplemented!("ambiguous obligations")
+            self.obligations
+                .drain(..)
+                .map(|obligation| FulfillmentError {
+                    obligation: obligation.clone(),
+                    code: FulfillmentErrorCode::CodeSelectionError(SelectionError::Unimplemented),
+                    root_obligation: obligation,
+                })
+                .collect()
         }
```
Comment (oli-obk) [inline compiler/rustc_trait_selection/src/solve/fulfill.rs:62]: I guess this `if` condition can now be removed and we can just use the else path?

**Review by oli-obk — CHANGES_REQUESTED** [review]

File: compiler/rustc_trait_selection/src/solve/fulfill.rs [hunk 2]
```diff
@@ -48,29 +51,46 @@ impl<'tcx> TraitEngine<'tcx> for FulfillmentCtxt<'tcx> {
         if self.obligations.is_empty() {
             Vec::new()
         } else {
-            unimplemented!("ambiguous obligations")
+            self.obligations
+                .drain(..)
+                .map(|obligation| FulfillmentError {
+                    obligation: obligation.clone(),
+                    code: FulfillmentErrorCode::CodeSelectionError(SelectionError::Unimplemented),
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/solve/fulfill.rs:58]: that shouldn't be unimplemented, should it? 
```suggestion
                    code: FulfillmentErrorCode::CodeAmbiguity,
```

**Review by lcnr — COMMENTED** [review]
nit, then r=me

**Pull request merged**